From 46b0df9e895ab5fe1af7e41f1ff770945a229f79 Mon Sep 17 00:00:00 2001
From: Shawn Xiao <b49994@freescale.com>
Date: Tue, 19 May 2015 14:32:46 +0800
Subject: [PATCH 1135/1594] MGS-622 [#1729] Suspend/resume test cause GPU hang

commit 4ac2e19117d3c2349cab50e58a2b33606c3d5bdd from
git://git.freescale.com/imx/linux-2.6-imx.git

Signal will cause some GPU kernel call return with error status
gcvSTATUS_INTERRUPTED. If adding interruptcount first, the signal will
cause the number not align with actual event, and the monitor funciton
will make a fake GPU hang judgement based on this.

Date May 21, 2015

Signed-off-by: Shawn Xiao <b49994@freescale.com>
---
 .../mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c   |   16 ++++++++--------
 1 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c
index f160ae9..ed539f6 100644
--- a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c
+++ b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c
@@ -1725,14 +1725,6 @@ gckEVENT_Submit(
             /* Determine cache needed to flush. */
             gcmkVERIFY_OK(_QueryFlush(Event, Event->queues[id].head, &flush));
 
-#if gcdINTERRUPT_STATISTIC
-            gcmkVERIFY_OK(gckOS_AtomIncrement(
-                Event->os,
-                Event->interruptCount,
-                &oldValue
-                ));
-#endif
-
 #if gcdNULL_DRIVER
             /* Notify immediately on infinite hardware. */
             gcmkONERROR(gckEVENT_Interrupt(Event, 1 << id));
@@ -1832,6 +1824,14 @@ gckEVENT_Submit(
             gcmkONERROR(gckCOMMAND_Execute(command, executeBytes));
 #endif
 #endif
+#if gcdINTERRUPT_STATISTIC
+            gcmkVERIFY_OK(gckOS_AtomIncrement(
+                Event->os,
+                Event->interruptCount,
+                &oldValue
+                ));
+#endif
+
         }
 
         /* Release the command queue. */
-- 
1.7.5.4

